File: crash-loop-sdk-app-ex/app/src/main/java/com/example/foo/FeatureMgr.java
package com.example.foo;

//Representative of an integrated Feature Flag Manager such as LaunchDarkly, Split.io, etc.
public class FeatureMgr {

    static boolean doWeCrash = true;
    static String FF = "new_launch_video";

    public static boolean turnoff(String msg) {
        if (msg == FF){
            //Set boolean to false so that the exception does not occur during launch.
            //Representative of turning off a feature flag whose code causes an exception at launch.
            doWeCrash = false;
        }
        return (doWeCrash);
    }
    public static String get_active_features() {
        return (FF);
    }
}

File: swift-ios/swift-ios/FeatureMgr.swift
//
//  FeatureMgr.swift
//  swift-ios
//
//  Created by Barbara Sudol on 6/9/22.
//  Copyright © 2022 Bugsnag. All rights reserved.
//

import Foundation


// Simple class allowing us to interact with a "Feature Manager" -> when we use the turn_off function to "turn off"
// our feature flags, we disable the code that was causing an exception, allowing the app to successfully launch.
class FeatureMgr {
    static var doWeCrash = true
    static var FF = "new_launch_video"
    
    static func turn_off(msg: String) -> Bool {
        if (msg == FF){
            doWeCrash = false
        }
        return doWeCrash
    }
    
    static func get_active_features() -> String { return FF }
    
    static func generateUncaughtException() {
        let someJson : Dictionary = ["foo":self]
        do {
            let data = try JSONSerialization.data(withJSONObject: someJson, options: .prettyPrinted)
            print("Received data: %@", data)
        } catch {
            // Why does this crash the app? A very good question.
        }
    }
}
